### ISSUE TITLE
Extend Shanq capabilities

### ISSUE THREAD

**Issue by FacticiusVir** [FacticiusVir/SharpVk#12]
- [ ] Support texture samplimg

**Comment by sunkin351** [comment]
Do you think you can make this capable of generating compute shaders? And are there any examples available?

**Comment by FacticiusVir** [comment]
Yep, compute shaders are a definite possibility; I'm working on porting some of Sascha's examples at the minute, will include the compute sample and see how much work would be required for Shanq support.

**Comment by sunkin351** [comment]
Off Topic question: If I were to make a proper OpenCL C shader, how would I map the buffers/variables to the shader through vulkan? Do you know? I can compile OpenCL C to spir-v, so that's not an issue.

**Comment by FacticiusVir** [comment]
I'm not that familiar with OpenCL/Vulkan, sorry; I'd probably look at the disassembly of the OpenCL SPIR-V vs. a GLSL compute shader and go from there.
